### ISSUE TITLE
Property descriptor differences are not tracked in branches

### ISSUE THREAD

**Issue by sebmarkbage** [facebook/prepack#1138]
I noticed this but I guess it is probably very unlikely to actually come up in practice.

```js
  var c = __abstract('boolean', 'c');
  var obj = {};
  if (c) {
    Object.defineProperty(obj, 'foo', {
      value: 123,
      enumerable: true
    });
  } else {
    Object.defineProperty(obj, 'foo', {
      enumerable: false,
      value: 123,
    });
  }
  result = obj;
```

This will only track enumerability for one of the branches and encode that.

**Comment by hermanventer** [comment]
#1116 should already address this.

**Comment by NTillmann** [comment]
Addressed by #1116.
